velox/functions/prestosql/types/QDigestRegistration.cpp

```diff
@@ -34,8 +35,20 @@ class QDigestTypeFactories : public CustomTypeFactories {
   }
 
   AbstractInputGeneratorPtr getInputGenerator(
-      const InputGeneratorConfig& /*config*/) const override {
-    return nullptr;
+      const InputGeneratorConfig& config) const override {
+    // QDigest supports three types: BIGINT, REAL, and DOUBLE
+    static const std::vector<TypePtr> kQDigestTypes{
+        BIGINT(),
+        REAL(),
+        DOUBLE(),
+    };
+    fuzzer::FuzzerGenerator rng(config.seed_);
+    auto index = boost::random::uniform_int_distribution<size_t>(
+        0, kQDigestTypes.size() - 1)(rng);
+    const auto& paramType = kQDigestTypes[index];
```

Comment:
We would need to generate input for the exact type instead of for a random qdigest type. The problem here is that getInputGenerator doesn't know the exact qdigest type being requested. To address this problem, you can add a TypePtr to InputGeneratorConfig, so that getInputGenerator() knows the exact data type. This TypePtr in InputGeneratorConfig can be set here: https://github.com/facebookincubator/velox/blob/0d58504b24b1640748e340a03bd3b5c4b7c1aa23/velox/vector/fuzzer/VectorFuzzer.cpp#L250